conversion.go

```diff
@@ -46,7 +102,41 @@ func convertFromNumber[NumOut Number, NumIn Number](orig NumIn) (converted NumOu
 	}
 }
 
-// ToInt attempts to convert any [Number] value to an int.
+func convertFromString[NumOut Number](s string) (converted NumOut, err error) {
+	s = strings.TrimSpace(s)
+
+	if b, err := strconv.ParseBool(s); err == nil {
+		if b {
+			return NumOut(1), nil
+		}
+		return NumOut(0), nil
+	}
+
+	if strings.Contains(s, ".") {
+		o, err := strconv.ParseFloat(s, 64)
+		if err != nil {
+			return 0, fmt.Errorf("%w: cannot convert %v to %T", ErrConversionIssue, s, converted)
+		}
+		return convertFromNumber[NumOut](o)
+	}
+
+	if s, found := strings.CutPrefix(s, "-"); found {
+		o, err := strconv.ParseInt(s, 0, 64)
+		if err != nil || o < 0 {
```

Comment:
Add a comment explaining the error returned by ParseInt is irrelevant

Also, the hack of removing the sign to make sure it won't overflow int

Also the check on o < 0, to avoid having string "--123" parsed as 123